addrmgr/addrmanager_test.go

```diff
@@ -558,3 +724,325 @@ func TestCorruptPeersFile(t *testing.T) {
+			valid:         true,
+			reach:         Ipv4,
+		},
+		{
+			name:          "routable ipv6 to routable ipv4",
+			localAddress:  routableIpv6Address,
+			remoteAddress: routableIpv4Address,
+			valid:         false,
+			reach:         Unreachable,
+		},
+		{
+			name:          "unroutable ipv4 to routable ipv6",
+			localAddress:  unroutableIpv4Address,
+			remoteAddress: routableIpv6Address,
+			valid:         false,
+			reach:         Default,
+		},
+		{
+			name:          "unroutable ipv6 to routable ipv6",
+			localAddress:  unroutableIpv6Address,
+			remoteAddress: routableIpv6Address,
+			valid:         false,
+			reach:         Default,
+		},
+		{
+			name:          "unroutable ipv4 to routable ipv6",
+			localAddress:  unroutableIpv4Address,
+			remoteAddress: routableIpv6Address,
+			valid:         false,
+			reach:         Default,
+		},
+		{
+			name:          "routable ipv4 to unroutable ipv6",
+			localAddress:  routableIpv4Address,
+			remoteAddress: unroutableIpv6Address,
+			valid:         false,
+			reach:         Unreachable,
+		},
+		{
+			name:          "routable ivp6 rfc4380 to routable ipv6",
+			localAddress:  rfc4380IPAddress,
+			remoteAddress: routableIpv6Address,
+			valid:         true,
+			reach:         Teredo,
+		},
+		{
+			name:          "routable ipv4 to routable ipv6",
+			localAddress:  routableIpv4Address,
+			remoteAddress: routableIpv6Address,
+			valid:         true,
+			reach:         Ipv4,
+		},
+		{
+			name:          "tunnelled ipv6 rfc3964 to routable ipv6",
+			localAddress:  rfc3964IPAddress,
+			remoteAddress: routableIpv6Address,
+			valid:         true,
+			reach:         Ipv6Weak,
+		},
+		{
+			name:          "tunnelled ipv6 rfc6052 to routable ipv6",
+			localAddress:  rfc6052IPAddress,
+			remoteAddress: routableIpv6Address,
+			valid:         true,
+			reach:         Ipv6Weak,
+		},
+		{
+			name:          "tunnelled ipv6 rfc6145 to routable ipv6",
+			localAddress:  rfc6145IPAddress,
+			remoteAddress: routableIpv6Address,
+			valid:         true,
+			reach:         Ipv6Weak,
+		},
+	}
+
+	addressManager := New("testValidatePeerNa", nil)
+	for _, test := range tests {
+		localIP := net.ParseIP(test.localAddress)
+		remoteIP := net.ParseIP(test.remoteAddress)
+		localNa := wire.NewNetAddressIPPort(localIP, 8333, wire.SFNodeNetwork)
+		remoteNa := wire.NewNetAddressIPPort(remoteIP, 8333, wire.SFNodeNetwork)
+
+		valid, reach := addressManager.ValidatePeerNa(localNa, remoteNa)
+		if valid != test.valid {
+			t.Errorf("%q: unexpected return value for valid - want '%v', "+
+				"got '%v'", test.name, test.valid, valid)
+			continue
+		}
+		if reach != test.reach {
+			t.Errorf("%q: unexpected return value for reach - want '%v', "+
+				"got '%v'", test.name, test.reach, reach)
+		}
+	}
+}
+
+// TestHostToNetAddress ensures that HostToNetAddress behaves as expected
+// given valid and invalid host name arguments.
+func TestHostToNetAddress(t *testing.T) {
```

Comment:
Rather than adding a test here, `HostToNetAddress` can be moved out of the address manager and moved to a private method on the main module along with these test cases. This also allows removing the need to pass in a `lookupFunc` to `New`.

Reply:
This does not seem necessary for this PR, deferring. Though if anyone has preference to see this change made here, adding it would not be an issue.